src/io.cpp

```diff
@@ -0,0 +1,97 @@
+#include <s2/s1angle.h>
+#include <s2geography.h>
+
+#include "constants.hpp"
+#include "creation.hpp"
+#include "geography.hpp"
+#include "pybind11.hpp"
+
+namespace py = pybind11;
+namespace s2geog = s2geography;
+using namespace spherely;
+
+class FromWKT {
+public:
+    FromWKT(bool oriented, bool planar, float tessellate_tol_m = 100.0) {
+#if defined(S2GEOGRAPHY_VERSION_MAJOR) && \
+    (S2GEOGRAPHY_VERSION_MAJOR >= 1 || S2GEOGRAPHY_VERSION_MINOR >= 2)
+        s2geog::geoarrow::ImportOptions options;
+        options.set_oriented(oriented);
+        if (planar) {
+            auto tol = S1Angle::Radians(tessellate_tol_m / EARTH_RADIUS_METERS);
+            options.set_tessellate_tolerance(tol);
+        }
+        m_reader = std::make_shared<s2geog::WKTReader>(options);
+#else
+        if (planar || oriented) {
+            throw std::invalid_argument(
+                "planar and oriented options are only available with s2geography >= 0.2");
+        }
+        m_reader = std::make_shared<s2geog::WKTReader>();
+#endif
+    }
+
+    PyObjectGeography operator()(py::str a) const {
+        return make_py_geography(m_reader->read_feature(a));
+    }
+
+private:
+    std::shared_ptr<s2geog::WKTReader> m_reader;
+};
+
+py::str to_wkt(PyObjectGeography a) {
+    s2geog::WKTWriter writer;
+    auto res = writer.write_feature(a.as_geog_ptr()->geog());
+    return py::str(res);
+}
+
+void init_io(py::module& m) {
+    m.def(
+        "from_wkt",
+        [](py::array_t<py::str> a, bool oriented, bool planar, float tessellate_tol_m) {
+            return py::vectorize(FromWKT(oriented, planar, tessellate_tol_m))(std::move(a));
+        },
+        py::arg("a"),
+        py::arg("oriented") = false,
+        py::arg("planar") = false,
+        py::arg("tessellate_tol_m") = 100.0,
+        R"pbdoc(
+        Creates geographies from the Well-Known Text (WKT) representation.
+
+        Parameters
+        ----------
+        a : str or array_like
+            WKT strings.
+        oriented : bool, default False
+            Set to True if polygon ring directions are known to be correct
+            (i.e., exterior rings are defined counter clockwise and interior
+            rings are defined clockwise).
+            By default (False), it will return the polygon with the smaller
+            area.
+        planar : bool, default False
+            If set to True, the edges of linestrings and polygons are assumed
+            to be linear on the plane. In that case, additional points will
+            be added to the line while creating the geography objects, to
+            ensure every point is within 100m of the original line.
+            By default (False), it is assumed that the edges are spherical
+            (i.e. represent the shortest path on the sphere between two points).
+        tessellate_tol_m : float, default 100.0
```

Comment:
```suggestion
        tessellate_tol : float, default 100.0
```

(nit suggestion: `_m` makes the parameter name slightly less readable IMHO and its meaning may not be obvious at a first glance without checking the docs anyway... `_meters` would be more meaningful but would also make the parameter name too long).

Reply:
That keyword name was taken from R s2 (https://r-spatial.github.io/s2/reference/s2_geog_point.html), but I am certainly fine with dropping the `_m` as well. I agree that from just seeing the keyword name (without reading the explanation for what it stands), it is not necessarily clear that this stands for "meters", making it an unnecessary suffix.

(to be honest, I am also not convinced that "tessellation" is the best name. BigQuery uses it, but if you google for it, it is typically about dividing a plane into multiple shapes that nicely cover it without overlap)

Reply:
Yes I'm more familiar with the latter meaning (dividing a surface into multiple shapes), but I'm fine with the name here.

Another name could be `densify_pts_tol` or `densify_tol` (pyproj uses [densify_pts](https://pyproj4.github.io/pyproj/dev/api/transformer.html#pyproj.transformer.Transformer.transform_bounds) for similar problems).

Reply:
Going to merge this, so I can fixup conflicts with the geoarrow PR. But happy to change the keyword name again in a quick follow-up.